mapper/orm-coordination-outbox-polling/src/main/java/org/hibernate/search/mapper/orm/coordination/outboxpolling/cfg/UuidDataType.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.mapper.orm.coordination.outboxpolling.cfg;
+
+import org.hibernate.dialect.Dialect;
+import org.hibernate.dialect.SQLServerDialect;
+import org.hibernate.search.util.common.annotation.Incubating;
+
+@Incubating
+public enum UuidDataType {
+	/**
+	 * Database specific. Some databases might not work correctly with a particular representation of UUID hence the
+	 * default provides a sensible value depending on which particular database is used. By default, a character representation
+	 * is used as characters should be overall supported by databases.
+	 * <p>
+	 * But for some databases, a different value is used. For instance, in the case of Microsoft SQL Server,
+	 * using a character representation might lead to undesired behavior in the form of deadlocks
+	 * so a binary representation is used by default.
+	 */
+	DEFAULT( "default" ) {
+		@Override
+		public String type(Dialect dialect) {
+			// See HSEARCH-4749 why MSSQL is treated differently
+			if ( dialect instanceof SQLServerDialect ) {
+				return BINARY.type( dialect );
+			}
+			return CHAR.type( dialect );
+		}
+	},
+
+	UUID( "uuid" ),
+
+	BINARY( "uuid-binary" ),
+
+	CHAR( "uuid-char" );
+
+	private final String externalRepresentation;
+
+	UuidDataType(String externalRepresentation) {
+		this.externalRepresentation = externalRepresentation;
+	}
+
+	public String externalRepresentation() {
+		return externalRepresentation;
+	}
```

Comment:
You will most likely need to introduce an `of()` method similar to [this](https://github.com/hibernate/hibernate-search/blob/5bfe116a9b5810e069d36c40e7ee86775dcc954b/backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/multitenancy/MultiTenancyStrategyName.java#L29-L37) so that we can configure this more easily in Quarkus.